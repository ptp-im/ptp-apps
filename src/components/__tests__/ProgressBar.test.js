import * as React from 'react';
import { View } from 'react-native';

import { act } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import ProgressBar from '../ProgressBar.tsx';

const layoutEvent = {
  nativeEvent: {
    layout: {
      width: 100,
    },
  },
};

it('renders progress bar with specific progress', () => {
  const tree = renderer.create(<ProgressBar progress={0.2} />);
  act(() => {
    tree.root.findByType(View).props.onLayout(layoutEvent);
  });

  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders hidden progress bar', () => {
  const tree = renderer.create(<ProgressBar progress={0.2} visible={false} />);
  act(() => {
    tree.root.findByType(View).props.onLayout(layoutEvent);
  });

  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders colored progress bar', () => {
  const tree = renderer.create(<ProgressBar progress={0.2} color='red' />);
  act(() => {
    tree.root.findByType(View).props.onLayout(layoutEvent);
  });

  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders indeterminate progress bar', () => {
  const tree = renderer.create(<ProgressBar indeterminate />);
  act(() => {
    tree.root.findByType(View).props.onLayout(layoutEvent);
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
