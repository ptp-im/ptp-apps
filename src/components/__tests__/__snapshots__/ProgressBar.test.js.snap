// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders colored progress bar 1`] = `
<View
  accessibilityRole="progressbar"
  accessibilityState={
    Object {
      "busy": true,
    }
  }
  accessibilityValue={
    Object {
      "max": 100,
      "min": 0,
      "now": 20,
    }
  }
  accessible={true}
  onLayout={[Function]}
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "rgba(255, 0, 0, 0.38)",
        "height": 4,
        "opacity": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "red",
          "flex": 1,
          "transform": Array [
            Object {
              "translateX": -50,
            },
            Object {
              "scaleX": 0.0001,
            },
          ],
          "width": 100,
        }
      }
    />
  </View>
</View>
`;

exports[`renders hidden progress bar 1`] = `
<View
  accessibilityRole="progressbar"
  accessibilityState={
    Object {
      "busy": false,
    }
  }
  accessibilityValue={
    Object {
      "max": 100,
      "min": 0,
      "now": 20,
    }
  }
  accessible={true}
  onLayout={[Function]}
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "rgba(103, 80, 164, 0.38)",
        "height": 4,
        "opacity": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(103, 80, 164, 1)",
          "flex": 1,
          "transform": Array [
            Object {
              "translateX": -50,
            },
            Object {
              "scaleX": 0.0001,
            },
          ],
          "width": 100,
        }
      }
    />
  </View>
</View>
`;

exports[`renders indeterminate progress bar 1`] = `
<View
  accessibilityRole="progressbar"
  accessibilityState={
    Object {
      "busy": true,
    }
  }
  accessibilityValue={Object {}}
  accessible={true}
  onLayout={[Function]}
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "rgba(103, 80, 164, 0.38)",
        "height": 4,
        "opacity": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(103, 80, 164, 1)",
          "flex": 1,
          "transform": Array [
            Object {
              "translateX": -50,
            },
            Object {
              "scaleX": 0.0001,
            },
          ],
          "width": 100,
        }
      }
    />
  </View>
</View>
`;

exports[`renders progress bar with specific progress 1`] = `
<View
  accessibilityRole="progressbar"
  accessibilityState={
    Object {
      "busy": true,
    }
  }
  accessibilityValue={
    Object {
      "max": 100,
      "min": 0,
      "now": 20,
    }
  }
  accessible={true}
  onLayout={[Function]}
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "rgba(103, 80, 164, 0.38)",
        "height": 4,
        "opacity": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(103, 80, 164, 1)",
          "flex": 1,
          "transform": Array [
            Object {
              "translateX": -50,
            },
            Object {
              "scaleX": 0.0001,
            },
          ],
          "width": 100,
        }
      }
    />
  </View>
</View>
`;
